docs/user/datastores/runtime_config.rst

```diff
@@ -240,6 +240,19 @@ This property controls how long simple feature serializers will be cached in mem
 reduce the memory footprint of your application, at the cost of increased processing time. The expiry is specified
 as a duration, e.g. ``10 minutes`` or ``1 hour``. The default is ``1 hour``.
 
+geomesa.sort.memory.threshold
++++++++++++++++++++++++++++++
+
+This property can be used to constrain the memory used to sort result sets. GeoMesa sorts results in the client
+process memory, since the supported back-end databases don't offer native ordering. To avoid having large
+result sets exceed the client memory capacity, a memory threshold can be set. Once the size of a result set
+exceeds this threshold, additional results will be written to disk and sorted there. Note that the actual memory
+used may exceed the threshold, as the memory footprint calculation is an estimation. The threshold is specified
+as a number of bytes, e.g. ``10MB``. The default is to always sort in memory.
+
+Note that distributed Arrow queries will never use disk to sort, due to the nature of Arrow result batches. For
+supported back-ends, sorting on disk for Arrow queries can be achieved by disabling remote Arrow processing.
```

Comment:
Later, I'm going to be looking for a warning if a user has set this setting and remote arrow queries are still enabled...